plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -102,6 +102,27 @@
 public class ThriftHiveMetastoreClient
         implements ThriftMetastoreClient
 {
+    /**
+     * This character is used to mark a database name as having a catalog name prepended.  This
+     * marker should be placed first in the String to make it easy to determine that this has both
+     * a catalog and a database name.  @ is chosen as it is not used in regular expressions.  This
+     * is only intended for use when making old Thrift calls that do not support catalog names.
+     */
+    private static final char CATALOG_DB_THRIFT_NAME_MARKER = '@';
+
+    /**
+     * This String is used to seaprate the catalog name from the database name.  This should only
+     * be used in Strings that are prepended with {@link #CATALOG_DB_THRIFT_NAME_MARKER}.  # is
+     * chosen because it is not used in regular expressions.  this is only intended for use when
+     * making old Thrift calls that do not support catalog names.
+     */
+    private static final String CATALOG_DB_SEPARATOR = "#";
+
+    /**
+     * Mark a database as being empty (as distinct from null).
+     */
+    private static final String DB_EMPTY_MARKER = "!";
```

Comment:
is this for the Hive 2?

Reply:
I adopted this approach from Hive-JDBC version 3.x. I believe it's their strategy for maintaining compatibility with Hive 2.x.